client/src/blocking.rs

```diff
@@ -71,12 +108,35 @@ impl<C: Deref<Target = impl Signer> + Clone> Program<C> {
 }
 
 impl<'a, C: Deref<Target = impl Signer> + Clone> RequestBuilder<'a, C> {
+    #[cfg(not(feature = "rpc-client"))]
+    pub fn from(
+        program_id: Pubkey,
+        cluster: &str,
+        payer: C,
+        options: Option<CommitmentConfig>,
+        handle: &'a Handle,
+    ) -> Self {
+        Self {
+            program_id,
+            payer,
+            cluster: cluster.to_string(),
+            accounts: Vec::new(),
+            options: options.unwrap_or_default(),
+            instructions: Vec::new(),
+            instruction_data: None,
+            signers: Vec::new(),
+            handle,
+        }
+    }
```

Comment:
It would be better to only put the new param behind the feature flag rather than duplicating the whole method.